### ISSUE TITLE
Add support for managing Emoji Categories

### ISSUE THREAD

**Issue by psydwannabe** [mastodon/mastodon#12134]
### Pitch
My pitch is some sort of rudimentary emoji category management interface. It would ideally be able to:
* Delete categories
* Rename categories
* List the number of emojis presently associated to each category

### Motivation
Presently, Emoji categories are created on demand from the Custom Emojis (local) page- and that's great! But there's no way to delete or rename categories, so if you misspell one, the misspelled category will be lingering in the list (admin only, thankfully) FOREVER. D: 

Also, because the list of categories does not include any indication of use, there's no way to tell from the UI which categories are in use and which aren't.

**Comment by Gargron** [comment]
Yes, this is much needed. Only in 3.1, however, this isn't something I can add as a quick fix.

**Comment by SJang1** [comment]
This is really needed :D

**Comment by SebastianBerlin** [comment]
Indeed, at least a remove-function for the categories would be helpful in the meantime.

**Comment by psydwannabe** [comment]
It'd be nice to have this feature some year.

**Comment by yuptogun** [comment]
here's a set of queries that theoretically purge all custom emoji categories.

```sql
-- never been tested
UPDATE custom_emojis SET category_id = NULL WHERE category_id IS NOT NULL;
TRUNCATE TABLE custom_emoji_categories;
```

if you have not much options, maybe you give it a go and let me know whether it works. hope this helps!

**Comment by nuztalgia** [comment]
Just tested the above queries on my new unlaunched instance (nothing to lose 😂) and they worked perfectly!

Here are the exact steps I took, in case anyone else who followed the [default setup instructions](https://docs.joinmastodon.org/admin/install/#setup) wants to try this out:

1. On your host machine, if you're logged in as `root`, switch to the `mastodon` user.
```
su - mastodon
```

2. Launch `psql` and connect to the `mastodon_production` database.
```
psql -d mastodon_production
```

3. Erase all `category_id` assignments for every row in the `custom_emojis` table.
```sql
UPDATE custom_emojis SET category_id = NULL WHERE category_id IS NOT NULL;
```

4. Erase all the rows in the `custom_emoji_categories` table (but keep the table itself).
```sql
TRUNCATE TABLE custom_emoji_categories;
```

5. Exit out of `psql`.
```
\q
```

Finally, open up your Mastodon instance in your browser and observe that of all your custom emojis are now `Uncategorized`. 🎉

Of course, you *could* add more nuance to those SQL queries if you don't want to completely delete all of your emoji categories... but for those who just want to start over from scratch like I did, the above comment was spot-on. Thank you, @yuptogun! 💜
